core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -637,7 +640,11 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
 
   private def initializeControllerContext() {
     // update controller cache with delete topic information
-    controllerContext.liveBrokers = zkClient.getAllBrokersInCluster.toSet
+    val curBrokerAndEpochs = zkClient.getAllBrokerAndEpochsInCluster
+    controllerContext.liveBrokers = curBrokerAndEpochs.map(_._1).toSet
+    // update broker epoch cache for all live brokers
+    curBrokerAndEpochs.foreach(e => controllerContext.brokerEpochs(e._1.id) = e._2)
+    info(s"Initialized broker epochs cache: ${controllerContext.brokerEpochs}")
```

Comment:
I think it is better to keep it info because the broker epoch information is as informative as the broker ids. We als log the broker ids in info log so we should keep it consistent.